src/team_lib/hier/xccl_hier_schedule.c

```diff
@@ -333,6 +334,10 @@ xccl_status_t hier_task_progress_handler(xccl_coll_task_t *task)
     assert(self->req);
     for (i = 0; (i < n_polls) && (task->state == XCCL_TASK_STATE_INPROGRESS); i++) {
         status = xccl_collective_test(self->req);
+        if (self->scratch != NULL) {
```

Comment:
should this be in the completed branch?

Reply:
yes! thanks
moved to wrong place during rebase